qiskit/tools/jupyter/version_table.py

```diff
@@ -33,19 +34,30 @@ def qiskit_version_table(self, line="", cell=None):
         """
         html = "<h3>Version Information</h3>"
         html += "<table>"
-        html += "<tr><th>Qiskit Software</th><th>Version</th></tr>"
+        html += "<tr><th>Software</th><th>Version</th></tr>"
 
-        packages = []
-        qver = qiskit.__qiskit_version__
+        packages = {}
 
-        for pkg in qver:
-            if qver[pkg]:
-                packages.append((f"<code>{pkg}</code>", qver[pkg]))
+        from importlib.metadata import metadata, PackageNotFoundError
 
-        for name, version in packages:
-            html += f"<tr><td>{name}</td><td>{version}</td></tr>"
+        try:
+            packages["qiskit"] = metadata("qiskit")["Version"]
+        except PackageNotFoundError:
+            packages["qiskit"] = None
+
+        packages["qiskit-terra"] = qiskit.__version__
+
+        qiskit_modules = {module.split(".")[0] for module in modules.keys() if "qiskit" in module}
+        for qiskit_module in qiskit_modules:
+            try:
+                packages[metadata(qiskit_module)["Name"]] = metadata(qiskit_module)["Version"]
+            except PackageNotFoundError:
+                packages["qiskit"] = None
```

Comment:
Looking at this with fresh eyes my concern here is that the module name in `sys.modules` isn't necessarily going to be the package name that importlib.metadata will find. The best example of this is actually `qiskit`,  if you just have terra installed you would need to call `metadata("qiskit-terra")` not `metadata("qiskit")` to get the installed package version for the `qiskit` module in `sys.modules`.

I think it might be better to do:

```suggestion
        qiskit_modules = {module.split(".")[0] for module in modules.keys() if "qiskit" in module}
        for qiskit_module in qiskit_modules:
            packages[qiskit_module] = getattr(sys.modules[qiskit_module], "__version__", None)
```

so it just uses the version reported in the `__version__` attribute.